flutter_bird_image_server/create_birds/generate_metadata.js

```diff
@@ -0,0 +1,66 @@
+const crypto = require('crypto');
+const traits = require("../input/traits.json");
+
+function getRandomWeightedTrait(traitList, guaranteed = false, rng) {
+    let totalWeight = traitList.reduce((sum, trait) => {
+        if (!guaranteed || (trait.name !== "none" && trait.name !== "default")) {
+            return sum + trait.weight;
+        }
+        return sum;
+    }, 0);
+
+    let randomNumber = rng() * totalWeight;
+    
+    for (const trait of traitList) {
+        if (guaranteed && (trait.name === "none" || trait.name === "default")) {
+            continue;
+        }
+        
+        randomNumber -= trait.weight;
+        if (randomNumber <= 0) {
+            return trait.name === "none" ? "" : trait.name;
+        }
+    }
+    
+    return "";
+}
+
+function generateMetadata(tokenId) {
+    console.log('Creating metadata for skin ' + tokenId);
+
+    const seed = crypto.createHash('sha256').update(tokenId.toString()).digest('hex');
+    const rng = require('seedrandom')(seed);
+
+    const skinTemplate = require('../input/metadata_template.json');
+    const skinMetadata = JSON.parse(JSON.stringify(skinTemplate));
+
+    const birdList = traits.bird;
+    const headList = traits.head;
+    const eyesList = traits.eyes;
+    const mouthList = traits.mouth;
+    const neckList = traits.neck;
+
+    let guaranteedTrait = Math.floor(rng() * 5) - 1;
+    const randomBird = getRandomWeightedTrait(birdList, false, rng);
+    if (randomBird === 'default') guaranteedTrait = Math.floor(rng() * 4);
+
+    let guaranteedNotTrait = Math.floor(rng() * 6);
+    if (guaranteedTrait === guaranteedNotTrait) guaranteedNotTrait = -1;
```

Comment:
What do guaranteedTrait and guaranteedNotTrait refer to?

Reply:
This variable determines how randomly each element of the randomly generated bird (head, mouth, etc.) will be generated.

guaranteedTrait is a variable that guarantees that a specific element will always be selected other than the default.
guaranteedNotTrait is a variable that guarantees that a specific element will be the default.